src/rpcmining.cpp

```diff
@@ -211,7 +211,7 @@ UniValue setgenerate(const UniValue& params, bool fHelp)
     if (params.size() > 0)
         fGenerate = params[0].get_bool();
 
-    int nGenProcLimit = -1;
+    int nGenProcLimit = GetArg("-genproclimit", DEFAULT_GENERATE_THREADS);
```

Comment:
Is it a bugfix or enhanced functionality? Again, why one uses GetArg() but the other doesn't?